groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -869,6 +992,39 @@ struct variantMoveFunction {
     }
 };
 
+template <class R, class VISITOR, class CVVARIANT, class TYPE, class CVTYPE>
```

Comment:
hm, this naming is only to allow for the name VARIANT inside the function. I wouldn't think it wise to start naming all parameters CVWHATSIT if they are CV qualified. I renamed them VARIANT and ARGTYPE in all instances in which they appear, and then I modified the function bodies to strip the cv qualifiers when calculating the index. For preview purposes, here is an example of one of the modified functions 

```
template <class R, class VISITOR, class VARIANT, class TYPE, class ARGTYPE>
struct variantFunctionId {
    static R functionImpl(VISITOR& visitor, VARIANT& variant)
    {
        static const size_t INDEX = BSLSTL_VARIANT_INDEX_OF(
            TYPE, typename bsl::remove_cv<VARIANT>::type);

        bsl::function<R(bsl::in_place_index_t<INDEX>, ARGTYPE&)> func(visitor);
        return func(bsl::in_place_index_t<INDEX>(), bsl::get<TYPE>(variant));
    }
};
```

Let me know if you prefer a different way of going about this.

Reply:
Yeah, I agree that using `VARIANT` consistently is better than using `CVVARIANT`.
One small suggestion: maybe it would help to rename `TYPE` to `ALT_TYPE`. (It seems that it denotes the alternative type as opposed to the argument type `ARGTYPE` which might be a cv-qualified version thereof.)

Reply:
bumping the "One small suggestion" above